src/bigtiffimage.cpp

```diff
@@ -107,17 +108,24 @@ namespace Exiv2
             }
             else
             {
-                byte buffer[8];
+                byte buffer[8] = {0, 0, 0, 0, 0, 0, 0, 0};
                 io.read(buffer, 2);
                 const int size = getUShort(buffer, byteOrder);
-                assert(size == 8);
 
-                io.read(buffer, 2); // null
+                if (size == 8)
+                {
+                    io.read(buffer, 2); // null
+
+                    io.read(buffer, 8);
+                    const uint64_t offset = getULongLong(buffer, byteOrder);
 
-                io.read(buffer, 8);
-                const uint64_t offset = getULongLong(buffer, byteOrder);
+                    if (offset >= io.size())
```

Comment:
Is `offset` the offset from the start of the file or from the current position?

Reply:
it is the offset from the start of the file